hibernate-core/src/main/java/org/hibernate/bytecode/enhance/internal/bytebuddy/ModelTypePool.java

```diff
@@ -92,6 +94,13 @@ protected Resolution doDescribe(final String name) {
 
 	@Override
 	public void registerClassNameAndBytes(final String className, final byte[] bytes) {
+		//Very important: ensure the registered override is actually effective in case this class
+		//was already resolved in the recent past; this could have happened for example as a side effect
+		//of symbol resolution during enhancement of a different class, or very simply when attempting
+		//to re-enhanced the same class - which happens frequently in WildFly because of the class transformers
+		//being triggered concurrently by multiple parallel deployments.
+		resolutions.remove( className );
+		poolCache.remove( className );
```

Comment:
Maybe do this in `deregisterClassNameAndBytes` as well? Just in case...

Reply:
just to highlight, I've ignored this comment as I'm not very comfortable with the "just in case" ..it's indeed tempting, but it's beind de-registered in a `finally` block already so I wonder if it's making things a bit confusing for "future us".
Feel free to add it later if you feel strongly about it.